File: spring-boot-dubbo-simple/spring-boot-dubbo-simple-consumer/src/main/java/com/hhjian/controller/UserController.java
package com.hhjian.controller;

import com.alibaba.boot.dubbo.annotation.DubboConsumer;
import com.hhjian.domain.User;
import com.hhjian.service.UserService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

/**
 * <p>用户控制层</p>
 *
 * @author <a href="mailto:hhjian.top@qq.com">hhjian</a>
 * @since 2017.09.13
 */
@RestController
public class UserController {

    @DubboConsumer
    private UserService userService;

    @GetMapping("/users/{id}")
    public User getUser(@PathVariable Long id) {
        User user = userService.getUser(id);
        return user;
    }

}

File: spring-boot-kotlin/src/main/kotlin/com/hhjian/kotlin/controller/UserController.kt
package com.hhjian.kotlin.controller

import com.hhjian.kotlin.domain.User
import com.hhjian.kotlin.query.UserInfoQuery
import com.hhjian.kotlin.repository.UserRepository
import org.springframework.beans.BeanUtils
import org.springframework.web.bind.annotation.*

/**
 * <p>User Controller</p>
 *
 * @author <a href="mailto:hhjian.top@qq.com">hhjian</a>
 * @since 2018.01.05
 */
@RestController
class UserController(private val userRepository: UserRepository) {

    @GetMapping("/users")
    fun listUser(): List<User> = userRepository.findAll()

    @GetMapping("/users/{userId}")
    fun getUser(@PathVariable userId: String): User = userRepository.findOne(userId)

    @PutMapping("/users/{userId}")
    fun updateUser(@PathVariable userId: String, @RequestBody userInfo: UserInfoQuery): User {
        val userDO: User = userRepository.findOne(userId)
        BeanUtils.copyProperties(userInfo, userDO)
        return userRepository.save(userDO)
    }
}
